copier/copier.go

```diff
@@ -0,0 +1,1367 @@
+			}
+			if excluded {
+				continue
+			}
+			// if the glob was an absolute path, reconstruct the
+			// path that we should hand back for the match
+			var resultName string
+			if filepath.IsAbs(request.preservedGlobs[i]) {
+				resultName = filepath.Join(request.rootPrefix, globbed)
+			} else {
+				relResult := rel
+				if request.Directory != request.Root {
+					relResult, err = filepath.Rel(request.Directory, globbed)
+					if err != nil {
+						return errorResponse("copier: stat: error making %q relative to %q: %v", globbed, request.Directory, err)
+					}
+				}
+				resultName = relResult
+			}
+			result := StatForItem{Name: resultName}
+			s.Globbed = append(s.Globbed, resultName)
+			s.Results[resultName] = &result
+			// lstat the matched value
+			linfo, err := os.Lstat(globbed)
+			if err != nil {
+				result.Error = err.Error()
+				continue
+			}
+			result.Size = linfo.Size()
+			result.Mode = linfo.Mode()
+			result.ModTime = linfo.ModTime()
+			result.IsDir = linfo.IsDir()
+			result.IsRegular = result.Mode.IsRegular()
+			result.IsSymlink = (linfo.Mode() & os.ModeType) == os.ModeSymlink
+			checkForArchive := request.StatOptions.CheckForArchives
+			if result.IsSymlink {
+				// if the match was a symbolic link, read it
+				immediateTarget, err := os.Readlink(globbed)
+				if err != nil {
+					result.Error = err.Error()
+					continue
+				}
+				// record where it points, both by itself (it
+				// could be a relative link) and in the context
+				// of the chroot
+				result.ImmediateTarget = immediateTarget
+				resolvedTarget, err := resolvePath(request.Root, globbed, pm)
+				if err != nil {
+					return errorResponse("copier: stat: error resolving %q: %v", globbed, err)
+				}
+				// lstat the thing that we point to
+				info, err := os.Lstat(resolvedTarget)
+				if err != nil {
+					result.Error = err.Error()
+					continue
+				}
+				// replace IsArchive/IsDir/IsRegular with info about the target
+				if info.Mode().IsRegular() && request.StatOptions.CheckForArchives {
+					result.IsArchive = isArchivePath(resolvedTarget)
+					checkForArchive = false
+				}
+				result.IsDir = info.IsDir()
+				result.IsRegular = info.Mode().IsRegular()
+			}
+			if result.IsRegular && checkForArchive {
+				// we were asked to check on this, and it
+				// wasn't a symlink, in which case we'd have
+				// already checked what the link points to
+				result.IsArchive = isArchivePath(globbed)
+			}
+		}
+		// no unskipped matches -> error
+		if len(s.Globbed) == 0 {
+			s.Globbed = nil
+			s.Results = nil
+			s.Error = fmt.Sprintf("copier: stat: %q: %v", glob, syscall.ENOENT)
+		}
+	}
+	return &Response{Stat: StatResponse{Globs: stats}}
+}
+
+func copierHandlerGet(bulkWriter io.Writer, request Request, pm *fileutils.PatternMatcher, idMappings *idtools.IDMappings) (*Response, func() error, error) {
+	statRequest := request
+	statRequest.Request = RequestStat
+	statResponse := copierHandlerStat(request, pm)
```

Comment:
AFAICS nothing at all ever uses this data, so, why? It’s expensive, it causes `Get` to set surprising `StatOptions`, and ultimately it doesn’t matter (apart from the possible reported error?).